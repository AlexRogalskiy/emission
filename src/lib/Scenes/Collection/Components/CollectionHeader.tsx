import { Box, color, Serif } from "@artsy/palette"
import OpaqueImageView from "lib/Components/OpaqueImageView/OpaqueImageView"
import React from "react"
import { Dimensions } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"

interface CollectionHeaderProps {
  collection: {
    title: string
    description: string
    headerImage: string
    image: {
      edges: Array<{
        node: {
          imageUrl: string
        }
      }>
    }
  }
}

export const CollectionHeader: React.SFC<CollectionHeaderProps> = props => {
  const { title, description, image } = props.collection
  const url = image.edges[0].node.imageUrl || ""
  const { width: screenWidth } = Dimensions.get("window")
  const imageHeight = 204

  return (
    <>
      <Box mb={2}>
        <OpaqueImageView imageURL={url} height={imageHeight} width={screenWidth} />
      </Box>
      <Serif size="8" color={color("black100")} ml={2}>
        {title}
      </Serif>
      <Serif size="4" color={color("black100")} mt={2}>
        {description}
      </Serif>
    </>
  )
}

export const CollectionHeaderContainer = createFragmentContainer(CollectionHeader, {
  collection: graphql`
    fragment CollectionHeader_collection on MarketingCollection {
      title
      description
      headerImage
      image: artworksConnection(sort: "-merchandisability", first: 1) {
        edges {
          node {
            imageUrl
          }
        }
      }
    }
  `,
})
