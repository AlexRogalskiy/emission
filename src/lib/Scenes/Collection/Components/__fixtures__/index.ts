export const CollectionFixture = {
  collection: {
    title: "Street Art Now",
    description:
      '<p>From <a href="https://www.someartsyimage.net/artist/jean-michel-basquiat">Jean-Michel Basquiat</a> to <a href="https://www.someartsyimage.net/artist/shepard-fairey">Shepard Fairey</a>, street artists are known for their creative expressions of rebellion.</p>',
    headerImage: "http://someimages.net/images/banksygirlwithballoon.png",
    image: { edges: [{ node: { imageUrl: "https://someimage.cloudfront.net/square.jpg" } }] },
  },
}
