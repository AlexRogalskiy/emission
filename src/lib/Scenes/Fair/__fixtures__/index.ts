export const fairFixture = {
  gravityID: "sofa-chicago-2018",
  internalID: "5aa1a9ee8b3b817c61dff02d",
  followed_content: null as any,
  name: "SOFA CHICAGO 2018",
  exhibition_period: "Nov 1 – 4, 2018",
  image: {
    image_url: "https://d32dm0rphc51dk.cloudfront.net/o0Zrtm-CgwbV1syD_33DuA/:version.jpg",
    aspect_ratio: 1,
    url: "https://d32dm0rphc51dk.cloudfront.net/o0Zrtm-CgwbV1syD_33DuA/square.jpg",
  },
  sponsoredContent: {
    activationText: "This fair will be incroyable. Come check out why.",
    pressReleaseUrl: "https://www.somefair.com",
  },
  profile: {
    gravityID: "art-basel-in-miami-beach-2018",
    internalID: "dfdlfjfua7d8a0_",
    is_followed: false,
    name: "art-basel",
    icon: {
      gravityID: "58ff95298b3b8104783c119c",
      href: null as any,
      height: 279,
      width: 279,
      url: "https://d32dm0rphc51dk.cloudfront.net/B1x-CqoWO1KitQnRj7PrrA/untouched-png.png",
    },
  },
  organizer: {
    profile: {
      icon: {
        gravityID: "58ff95298b3b8104783c119c",
        href: "",
        height: 279,
        width: 279,
        url: "https://d32dm0rphc51dk.cloudfront.net/B1x-CqoWO1KitQnRj7PrrA/untouched-png.png",
      },
      name: "SOFA CHICAGO",
      id: "UHJvZmlsZTpzb2ZhLWNoaWNhZ28=",
    },
    website: "",
  },
  start_at: "2018-11-01T17:00:00+00:00",
  end_at: "2018-11-04T18:00:00+00:00",
  formattedOpeningHours: "Opens Nov 1 at 5:00pm",
  location: {
    id: "TG9jYXRpb246NWJiNjJmMzYyYjVlN2QwMDI5ODliZDhh",
    gravityID: "5bb62f362b5e7d002989bd8a",
    address_2: "",
    postal_code: "60611",
    summary: "Festival Hall, Navy Pier \r\n600 East Grand Avenue Chicago, IL 60611",
    address: "",
    city: "Chicago",
    coordinates: {
      lat: 25.7949009,
      lng: -80.134698,
    },
    cached: null as any,
    day_schedules: null as any,
    day_schedule_text: null as any,
    openingHours: {
      schedules: [
        {
          days: "Monday–Sunday",
          hours: "Closed",
        },
      ],
    },
    display: "25.794890, -80.134550",
  },
  counts: {
    artists: 809,
    artworks: 2028,
    partners: 127,
    partner_shows: 127,
  },
  exhibitors_grouped_by_name: [
    {
      letter: "3",
      exhibitors: [
        {
          name: "303 Gallery",
          gravityID: "303 Gallery",
          profile_id: "303 Gallery",
        },
      ],
    },
    {
      letter: "A",
      exhibitors: [
        { name: "Acquavella Galleries", gravityID: "Acquavella Galleries", profile_id: "Acquavella Galleries" },
        { name: "Alan Cristea Gallery", gravityID: "Alan Cristea Gallery", profile_id: "Alan Cristea Gallery" },
        { name: "Alfonso Artiaco", gravityID: "Alfonso Artiaco", profile_id: "Alfonso Artiaco" },
        { name: "Altman Siegel", gravityID: "Altman Siegel", profile_id: "Altman Siegel" },
        { name: "Anat Ebgi", gravityID: "Anat Ebgi", profile_id: "Anat Ebgi" },
        { name: "Annely Juda Fine Art", gravityID: "Annely Juda Fine Art", profile_id: "Annely Juda Fine Art" },
        { name: "Anton Kern Gallery", gravityID: "Anton Kern Gallery", profile_id: "Anton Kern Gallery" },
      ],
    },
    {
      letter: "B",
      exhibitors: ["Bergamin & Gomide", "Berggruen Gallery", "Blum & Poe", "Boers-Li Gallery", "Bortolami"],
    },
    {
      letter: "C",
      exhibitors: [
        "CARDI GALLERY",
        "Casa Triângulo",
        "Casey Kaplan",
        "Cheim & Read",
        "Cristina Guerra Contemporary Art",
        "Crown Point Press",
      ],
    },
    {
      letter: "D",
      exhibitors: ["Dan Galeria", "DC Moore Gallery"],
    },
    {
      letter: "E",
      exhibitors: ["Edward Tyler Nahem Fine Art LLC", "Edwynn Houk Gallery", "Eric Firestone Gallery"],
    },
    {
      letter: "F",
      exhibitors: ["Fergus McCaffrey", "Fortes D'Aloia & Gabriel", "Francis M. Naumann Fine Art"],
    },
    {
      letter: "G",
      exhibitors: [
        "Gagosian",
        "Galerie Chantal Crousel",
        "Galerie Christophe Gaillard",
        "galerie frank elbaz",
        "GALERIE GEORGES-PHILIPPE ET NATHALIE VALLOIS",
        "Galerie Greta Meert",
        "Galerie Jérôme Poggi",
        "GALLERIA CONTINUA",
        "Gemini G.E.L.",
        "Gladstone Gallery",
        "Goodman Gallery",
        "GRIMM",
      ],
    },
    {
      letter: "H",
      exhibitors: ["Haines Gallery", "Hales Gallery", "Hammer Galleries", "Hauser & Wirth", "Howard Greenberg Gallery"],
    },
    {
      letter: "I",
      exhibitors: ["Ingleby Gallery", "Inman Gallery", "Instituto de Visión"],
    },
    {
      letter: "J",
      exhibitors: ["James Cohan"],
    },
    {
      letter: "K",
      exhibitors: ["Kasmin", "Kavi Gupta", "Kayne Griffin Corcoran", "KEWENIG"],
    },
    {
      letter: "M",
      exhibitors: ["Marian Goodman Gallery", "Marianne Boesky Gallery"],
    },
    {
      letter: "P",
      exhibitors: ["Peter Blum Gallery", "Pilar Corrias Gallery"],
    },
    {
      letter: "R",
      exhibitors: ["Rhona Hoffman Gallery", "Richard Gray Gallery"],
    },
    {
      letter: "S",
      exhibitors: ["Sean Kelly Gallery"],
    },
    {
      letter: "X",
      exhibitors: ["Xavier Hufkens"],
    },
  ],
  filteredArtworks: {
    aggregations: [
      {
        slice: "PRICE_RANGE",
        counts: [
          {
            id: "*-*",
            name: "for Sale",
          },
          {
            id: "1000-5000",
            name: "between $1,000 & $5,000",
          },
          {
            id: "5000-10000",
            name: "between $5,000 & $10,000",
          },
          {
            id: "10000-50000",
            name: "between $10,000 & $50,000",
          },
          {
            id: "*-1000",
            name: "Under $1,000",
          },
          {
            id: "50000-*",
            name: "over $50,000",
          },
        ],
      },
      {
        slice: "MEDIUM",
        counts: [
          {
            id: "sculpture",
            name: "Sculpture",
          },
          {
            id: "painting",
            name: "Painting",
          },
          {
            id: "design",
            name: "Design",
          },
          {
            id: "photography",
            name: "Photography",
          },
          {
            id: "jewelry",
            name: "Jewelry",
          },
          {
            id: "work-on-paper",
            name: "Work on Paper",
          },
          {
            id: "installation",
            name: "Installation",
          },
          {
            id: "prints",
            name: "Prints",
          },
          {
            id: "drawing",
            name: "Drawing",
          },
          {
            id: "film-slash-video",
            name: "Film/Video",
          },
        ],
      },
    ],
    artworks_connection: {
      edges: [
        {
          node: {
            gravityID: "watanabe-chiaki-links",
            title: "Links",
            date: "2018",
            sale_message: "Sold",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null as any,
            sale_artwork: null as any,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/uF3UWRba84Q4wSrRB-BxKA/large.jpg",
              aspect_ratio: 1.61,
            },
            artists: [
              {
                name: "Watanabe Chiaki",
              },
            ],
            partner: {
              name: "TAI Modern",
            },
            href: "/artwork/watanabe-chiaki-links",
          },
        },
        {
          node: {
            gravityID: "dylan-martinez-water-bag-number-34",
            title: "Water Bag #34",
            date: "2018",
            sale_message: "$1,800",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null as any,
            sale_artwork: null as any,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/EB-WA2El6g-tII076F5vFw/large.jpg",
              aspect_ratio: 0.67,
            },
            artists: [
              {
                name: "Dylan Martinez",
              },
            ],
            partner: {
              name: "Lyons Wier Gallery",
            },
            href: "/artwork/dylan-martinez-water-bag-number-34",
          },
        },
        {
          node: {
            gravityID: "dylan-martinez-water-bag-number-23",
            title: "Water Bag #23",
            date: "2018",
            sale_message: "$1,800",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null as any,
            sale_artwork: null as any,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/sAKQugZCVUZ9g5AoVyCJdA/large.jpg",
              aspect_ratio: 0.67,
            },
            artists: [
              {
                name: "Dylan Martinez",
              },
            ],
            partner: {
              name: "Lyons Wier Gallery",
            },
            href: "/artwork/dylan-martinez-water-bag-number-23",
          },
        },
        {
          node: {
            gravityID: "dylan-martinez-water-bag-number-32",
            title: "Water Bag #32",
            date: "2018",
            sale_message: "$1,800",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null as any,
            sale_artwork: null as any,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/jVS6Bolng37bN8GB0y8qfQ/large.jpg",
              aspect_ratio: 0.67,
            },
            artists: [
              {
                name: "Dylan Martinez",
              },
            ],
            partner: {
              name: "Lyons Wier Gallery",
            },
            href: "/artwork/dylan-martinez-water-bag-number-32",
          },
        },
        {
          node: {
            gravityID: "dylan-martinez-mach-2",
            title: "Mach 2",
            date: "2018",
            sale_message: "$12,000",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null as any,
            sale_artwork: null as any,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/tnjfaBzOFmFvfj227QjLJg/large.jpg",
              aspect_ratio: 1.5,
            },
            artists: [
              {
                name: "Dylan Martinez",
              },
            ],
            partner: {
              name: "Lyons Wier Gallery",
            },
            href: "/artwork/dylan-martinez-mach-2",
          },
        },
        {
          node: {
            gravityID: "dylan-martinez-water-bag-trio",
            title: "Water Bag Trio",
            date: "2018",
            sale_message: "$5,400",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null as any,
            sale_artwork: null as any,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/BwEuuw_gYmFs69ENusdang/large.jpg",
              aspect_ratio: 0.67,
            },
            artists: [
              {
                name: "Dylan Martinez",
              },
            ],
            partner: {
              name: "Lyons Wier Gallery",
            },
            href: "/artwork/dylan-martinez-water-bag-trio",
          },
        },
      ],
    },
  },
  artists: {
    edges: [
      {
        node: {
          href: "/tamas",
          gravityID: "tamas-abel",
          internalID: "1232222245",
          id: "QXJ0aXN0OnRhbWFzLWFiZWw=",
          name: "Tamás Ábel",
          is_followed: false,
          nationality: "Hungarian",
          birthday: "",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/mSFWvVbcQezxjjS0ordckg/large.jpg",
          },
        },
      },
      {
        node: {
          href: "/charlie",
          gravityID: "charlie-aldrich",
          internalID: "12347775",
          id: "QXJ0aXN0OmNoYXJsaWUtYWxkcmljaA==",
          name: "Charlie Aldrich",
          is_followed: false,
          nationality: "United States",
          birthday: "",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/3v-35X7zWI3p6Lk8p6pYhg/large.jpg",
          },
        },
      },
      {
        node: {
          href: "/rik",
          gravityID: "rik-allen",
          internalID: "123477750000000000",
          id: "QXJ0aXN0OnJpay1hbGxlbg==",
          name: "Rik Allen",
          is_followed: false,
          nationality: "",
          birthday: "",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/8Irjs-pCGNU9VvWg9r8wFg/tall.jpg",
          },
        },
      },
      {
        node: {
          href: "/jean",
          gravityID: "jean-luc-almond",
          internalID: "1232222245111",
          id: "QXJ0aXN0OmplYW4tbHVjLWFsbW9uZA==",
          name: "Jean-Luc Almond",
          is_followed: false,
          nationality: "English",
          birthday: "",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/qDxLZGl8BHeP8zCkoeW8tw/tall.jpg",
          },
        },
      },
      {
        node: {
          href: "/amador",
          gravityID: "amador",
          internalID: "123222224588888",
          id: "QXJ0aXN0OmFtYWRvcg==",
          name: "Amador",
          is_followed: false,
          nationality: "Spanish",
          birthday: "1957",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/UEPyf29fi9y4wwaHTJ9nqg/large.jpg",
          },
        },
      },
    ],
  },
  id: "RmFpcjpzb2ZhLWNoaWNhZ28tMjAxOA==",
  shows_connection: {
    edges: [
      {
        node: {
          gravityID: "spruth-magers",
          partner: {
            __typename: "Partner",
            profile: {
              gravityID: "spruth-magers",
              href: "/spruth-magers",
              name: "Sprüth Magers",
              internalID: "123456",
            },
          },
        },
      },
      {
        node: {
          gravityID: "cardi-gallery",
          partner: {
            __typename: "Partner",
            profile: {
              gravityID: "cardi-gallery",
              href: "/cardi-gallery",
              name: "CARDI GALLERY",
              internalID: "1234562",
            },
          },
        },
      },
    ],
  },
  shows: {
    pageInfo: {
      endCursor: "0.7605261301690746:5ba9f83787830600062e7f86",
      startCursor: "YXJyYXljb25uZWN0aW9uOjA=",
      hasNextPage: true,
      hasPreviousPage: false,
    },
    edges: [
      {
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          gravityID: "steidel-contemporary-steidel-contemporary-at-sofa-chicago-2018",
          internalID: "111",
          counts: {
            artworks: 4,
          },
          name: "Steidel Contemporary at SOFA CHICAGO 2018",
          is_fair_booth: true,
          partner: {
            id: "UGFydG5lcjpzdGVpZGVsLWNvbnRlbXBvcmFyeQ==",
            internalID: "123123123123123123123",
            gravityID: "steidel-contemporary",
            name: "Steidel Contemporary",
            href: "http://www.stei.com",
            profile: {
              is_followed: false,
              internalID: "1231231231",
            },
          },
          fair: {
            name: "SOFA CHICAGO 2018",
            id: "RmFpcjpzb2ZhLWNoaWNhZ28tMjAxOA==",
          },
          cover_image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/nltwPbizCU7VKKrjmEiqUA/medium.jpg",
          },
          location: {
            display: "Booth B32",
            id: "TG9jYXRpb246NWI5ODIwNjU0OTUzZjEwMDA1ZjY5ZTBh",
          },
          profile: {
            gravityID: "art-basel-in-miami-beach-2018",
          },
          artworks_connection: {
            edges: [
              {
                node: {
                  gravityID: "christine-flynn-costa-rica-palms-1",
                  title: "Costa Rica Palms",
                  date: "2018",
                  sale_message: "£2,500 - 5,000",
                  is_in_auction: false,
                  is_biddable: false,
                  is_acquireable: false,
                  is_offerable: false,
                  sale: null as any,
                  sale_artwork: null as any,
                  href: "/artwork/christine-flynn-costa-rica-palms-1",
                  artists: [
                    {
                      gravityID: "christine-flynn-1",
                      name: "Christine Flynn",
                    },
                  ],
                  partner: {
                    gravityID: "kahn-gallery",
                    internalID: "58d3ac7dc9dc241448f6ec7c",
                    name: "Kahn Gallery",
                  },
                  image: {
                    url: "https://d32dm0rphc51dk.cloudfront.net/sxzp6tLM3zFJ3dz5kfIqVA/larger.jpg",
                    aspect_ratio: 1.64,
                  },
                },
              },
              {
                node: {
                  gravityID: "reinhard-gorner-palacio-nacional-de-mafra-portugal",
                  title: "Palacio Nacional de Mafra, Portugal",
                  date: "2016",
                  sale_message: "$5,500",
                  is_in_auction: false,
                  is_biddable: false,
                  is_acquireable: false,
                  is_offerable: false,
                  sale: null as any,
                  sale_artwork: null as any,
                  href: "/artwork/reinhard-gorner-palacio-nacional-de-mafra-portugal",
                  artists: [
                    {
                      gravityID: "reinhard-gorner",
                      name: "Reinhard Gorner",
                    },
                  ],
                  partner: {
                    gravityID: "kahn-gallery",
                    internalID: "58d3ac7dc9dc241448f6ec7c",
                    name: "Kahn Gallery",
                  },
                  image: {
                    url: "https://d32dm0rphc51dk.cloudfront.net/nkFRIEsLDAZPUQkcN7Nb3Q/small.jpg",
                    aspect_ratio: 0.88,
                  },
                },
              },
              {
                node: {
                  gravityID: "elisabeth-lecourt-ma-poupee-de-cire-en-fanfaron-nepal",
                  title: "Ma Poupée de Cire en Fanfaron Népal",
                  date: "2018",
                  sale_message: "Sold",
                  is_in_auction: false,
                  is_biddable: false,
                  is_acquireable: false,
                  is_offerable: false,
                  sale: null as any,
                  sale_artwork: null as any,
                  href: "/artwork/elisabeth-lecourt-ma-poupee-de-cire-en-fanfaron-nepal",
                  artists: [
                    {
                      gravityID: "elisabeth-lecourt",
                      name: "Elisabeth LECOURT",
                    },
                  ],
                  partner: {
                    gravityID: "kahn-gallery",
                    internalID: "58d3ac7dc9dc241448f6ec7c",
                    name: "Kahn Gallery",
                  },
                  image: {
                    url: "https://d32dm0rphc51dk.cloudfront.net/GJPv6LnC3XpZasK1C3rzmQ/larger.jpg",
                    aspect_ratio: 0.74,
                  },
                },
              },
              {
                node: {
                  gravityID: "damien-gard-au-travers-des-siecles-1",
                  title: "Au Travers des Siècles",
                  date: "2017",
                  sale_message: "£2,500 - 5,000",
                  is_in_auction: false,
                  is_biddable: false,
                  is_acquireable: false,
                  is_offerable: false,
                  sale: null as any,
                  sale_artwork: null as any,
                  href: "/artwork/damien-gard-au-travers-des-siecles-1",
                  artists: [
                    {
                      gravityID: "damien-gard",
                      name: "Damien Gard",
                    },
                  ],
                  partner: {
                    gravityID: "kahn-gallery",
                    internalID: "58d3ac7dc9dc241448f6ec7c",
                    name: "Kahn Gallery",
                  },
                  image: {
                    url: "https://d32dm0rphc51dk.cloudfront.net/SdvT-flDhUHGblmIDdsqtg/medium_rectangle.jpg",
                    aspect_ratio: 0.71,
                  },
                },
              },
            ],
          },
          id: "U2hvdzpzdGVpZGVsLWNvbnRlbXBvcmFyeS1zdGVpZGVsLWNvbnRlbXBvcmFyeS1hdC1zb2ZhLWNoaWNhZ28tMjAxOA==",
          __typename: "Show",
        },
      },
    ],
  },
  isActive: true,
  hours:
    "Thursday, November 1, 2018: 5 - 9 pm\r\nFriday, November 2, 2018: 11 am - 7 pm (10 - 11 am VIP hour)\r\nSaturday, November 3, 2018: 11 am - 7 pm (10 - 11 am VIP hour)\r\nSunday, November 4, 2018: 12 - 6 pm ",
  links:
    "[Palm Beach Modern + Contempoary](http://www.artpbfair.com/)\r\n[Facebook](https://www.facebook.com/PalmBeachModernandContemporary) | [Instagram](https://www.instagram.com/artmiamifairs/) | [Twitter](https://twitter.com/PB_Modern)",
  about:
    "Cillum adipisicing dolore qui ullamco deserunt ut in. In ad do laboris fugiat nostrud. Officia et duis sunt velit cupidatat irure.\r\n\r\nAdipisicing aliqua sunt magna nisi qui mollit non amet commodo ad ex eu. Nostrud nostrud excepteur sint esse occaecat qui et magna reprehenderit. Non do eiusmod ullamco consequat.\r\n\r\nCommodo exercitation aliqua nostrud minim sint laboris et. Fugiat elit labore magna Lorem aliqua ex nulla ea in sunt consequat. Fugiat tempor minim ex ex ullamco quis ullamco ea. Proident consectetur pariatur aliqua minim voluptate ad sunt sint aliqua mollit ullamco enim proident. Aliquip adipisicing elit ad mollit ad magna est officia nisi mollit eu. Cupidatat aute ex deserunt ad voluptate labore ipsum voluptate. Elit proident nisi labore deserunt cupidatat exercitation dolor nulla sint mollit in.",
  ticketsLink: "https://ticket.artbasel.com/MCH/miamibeach2018/Shop",
}
