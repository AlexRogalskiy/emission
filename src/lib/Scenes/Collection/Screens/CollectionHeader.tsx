import { Box, color, Serif } from "@artsy/palette"
import { CollectionHeader_collection } from "__generated__/CollectionHeader_collection.graphql"
import OpaqueImageView from "lib/Components/OpaqueImageView/OpaqueImageView"
import { ReadMore } from "lib/Components/ReadMore"
import { truncatedTextLimit } from "lib/Scenes/Artwork/hardware"
import { Schema } from "lib/utils/track"
import React from "react"
import { Dimensions } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"

interface CollectionHeaderProps {
  collection: CollectionHeader_collection
}

const HEADER_IMAGE_HEIGHT = 204

export const CollectionHeader: React.SFC<CollectionHeaderProps> = props => {
  const { title, image, headerImage, descriptionMarkdown: collectionDescription } = props.collection
  const defaultHeaderUrl = image?.edges[0]?.node?.image?.resized?.url || ""
  const url = headerImage ? headerImage : defaultHeaderUrl
  const { width: screenWidth } = Dimensions.get("window")
  const textLimit = truncatedTextLimit()

  return (
    <>
      <Box mb={2}>
        <OpaqueImageView imageURL={url} height={HEADER_IMAGE_HEIGHT} width={screenWidth} useRawURL />
      </Box>
      <Box mb={collectionDescription ? 0 : 2}>
        <Serif size="8" color={color("black100")} ml={2}>
          {title}
        </Serif>
      </Box>
      {collectionDescription && (
        <Box m="2">
          <ReadMore
            content={collectionDescription}
            maxChars={textLimit}
            contextModule={Schema.ContextModules.Collection}
            trackingFlow={Schema.OwnerEntityTypes.Collection}
          />
        </Box>
      )}
    </>
  )
}

export const CollectionHeaderContainer = createFragmentContainer(CollectionHeader, {
  collection: graphql`
    fragment CollectionHeader_collection on MarketingCollection
      @argumentDefinitions(
        screenWidth: { type: "Int", defaultValue: 500 }
        headerHeight: { type: "Int", defaultValue: 204 }
      ) {
      title
      headerImage
      descriptionMarkdown
      image: artworksConnection(sort: "-decayed_merch", first: 1) {
        edges {
          node {
            image {
              resized(width: $screenWidth, height: $headerHeight) {
                url
              }
            }
          }
        }
      }
    }
  `,
})
