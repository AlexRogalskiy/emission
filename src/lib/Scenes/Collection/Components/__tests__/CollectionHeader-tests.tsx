import { mount } from "enzyme"
import OpaqueImageView from "lib/Components/OpaqueImageView/OpaqueImageView"
import React from "react"
import * as renderer from "react-test-renderer"
import { CollectionFixture } from "../__fixtures__"
import { CollectionHeader } from "../CollectionHeader"

describe("CollectionHeader", () => {
  it("Renders collection header correctly", () => {
    const component = renderer.create(<CollectionHeader {...CollectionFixture} />)
    expect(component).toMatchSnapshot()
  })

  it("passes an image url to collection header", () => {
    const wrapper = mount(<CollectionHeader {...CollectionFixture} />)
    expect(wrapper.find(OpaqueImageView).html()).toContain("https://someimage.cloudfront.net/square.jpg")
  })
})
