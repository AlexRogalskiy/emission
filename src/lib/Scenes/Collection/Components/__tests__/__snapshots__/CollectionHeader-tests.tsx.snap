// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionHeader Renders collection header correctly 1`] = `
Array [
  <View
    mb={2}
    style={
      Array [
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    <AROpaqueImageView
      height={204}
      imageURL="https://someimage.cloudfront.net/square.jpg"
      width={750}
    />
  </View>,
  <Text
    color="#000"
    fontFamily="ReactNativeAGaramondPro-Regular"
    fontSize="32px"
    lineHeight="38px"
    ml={2}
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "ReactNativeAGaramondPro-Regular",
          "fontSize": 32,
          "lineHeight": 38,
          "marginLeft": 8,
        },
        Object {},
      ]
    }
  >
    Street Art Now
  </Text>,
  <Text
    color="#000"
    fontFamily="ReactNativeAGaramondPro-Regular"
    fontSize="18px"
    lineHeight="26px"
    mt={2}
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "ReactNativeAGaramondPro-Regular",
          "fontSize": 18,
          "lineHeight": 26,
          "marginTop": 8,
        },
        Object {},
      ]
    }
  >
    &lt;p&gt;From &lt;a href="https://www.someartsyimage.net/artist/jean-michel-basquiat"&gt;Jean-Michel Basquiat&lt;/a&gt; to &lt;a href="https://www.someartsyimage.net/artist/shepard-fairey"&gt;Shepard Fairey&lt;/a&gt;, street artists are known for their creative expressions of rebellion.&lt;/p&gt;
  </Text>,
]
`;
